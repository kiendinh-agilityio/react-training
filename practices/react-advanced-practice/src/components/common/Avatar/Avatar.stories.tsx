import { Meta, StoryObj } from '@storybook/react';
import Avatar from '.';

export default {
  title: 'Avatar',
  component: Avatar,
  parameters: {
    layout: 'centered',
  },
} as Meta;

export const User: StoryObj = {
  args: {
    src: 'https://phimtat.vn/wp-content/uploads/2023/11/avt-3d.jpg',
  },
  decorators: [
    (Story) => (
      <div
        style={{
          width: '40px',
          height: '40px',
          borderRadius: '12px',
          overflow: 'hidden',
        }}
      >
        <Story />
      </div>
    ),
  ],
};
