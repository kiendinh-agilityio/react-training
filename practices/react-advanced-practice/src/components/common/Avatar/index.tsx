// import radix ui
import { Avatar as AvatarBase, AvatarProps } from '@radix-ui/themes';

type AvatarBaseProps = {
  ariaLabel?: string;
  alt?: string;
} & AvatarProps;

const Avatar = ({
  ariaLabel = 'Avatar',
  alt = 'Avatar',
  src = '',
  className = '',
  fallback,
}: AvatarBaseProps) => (
  <AvatarBase
    aria-label={ariaLabel}
    alt={alt}
    src={src}
    fallback={fallback}
    className={`rounded-base w-[40px] h-[40px] ${className}`}
  />
);

export default Avatar;
