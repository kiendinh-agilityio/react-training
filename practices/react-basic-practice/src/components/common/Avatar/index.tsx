interface AvatarProps {
  width: string;
  height: string;
  src: string;
  alt: string;
}

const Avatar = ({ width, height, src, alt }: AvatarProps) => (
  <img className={`rounded-base mr-3.5`} src={src} alt={alt} width={width} height={height} />
);

export default Avatar;
