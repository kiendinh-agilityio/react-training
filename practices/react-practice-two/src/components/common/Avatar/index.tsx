// import types
import { AvatarSize } from '@/types';

interface AvatarProps {
  url: string;
  alt: string;
  size?: AvatarSize;
}

const Avatar = ({ url, alt, size = AvatarSize.md }: AvatarProps) => (
  <img className={`${size} rounded-full`} src={url} alt={alt} />
);

export default Avatar;
