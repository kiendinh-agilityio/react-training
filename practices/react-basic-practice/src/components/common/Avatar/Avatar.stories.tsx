import Avatar from '.';

export default {
  component: Avatar,
  title: 'Avatar',
  tags: ['autodocs'],
  parameters: {
    layout: 'centered',
  },
};

export const User = {
  args: {
    src: 'https://phimtat.vn/wp-content/uploads/2023/11/avt-3d.jpg',
    alt: 'Esthera Jackson',
    width: '60px',
    height: '60px',
  },
};
