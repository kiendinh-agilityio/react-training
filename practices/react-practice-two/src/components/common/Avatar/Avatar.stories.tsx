import Avatar from '.';

export default {
  component: Avatar,
  title: 'Avatar',
  tags: ['autodocs'],
  parameters: {
    layout: 'centered',
  },
};

export const AvatarProfile = {
  args: {
    url: 'images/SerhiyHipskyy.png',
    alt: 'Serhiy Hipskyy',
  },
};
