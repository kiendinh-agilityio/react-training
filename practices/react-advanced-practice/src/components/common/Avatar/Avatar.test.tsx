import { render } from '@testing-library/react';

// Components
import Avatar from '.';

describe('Avatar component', () => {
  it('should render with A fallback', () => {
    const { container } = render(<Avatar fallback={'A'} />);
    expect(container).toMatchSnapshot();
  });
});
