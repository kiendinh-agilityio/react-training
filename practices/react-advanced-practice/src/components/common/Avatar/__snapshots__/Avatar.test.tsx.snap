// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar component should render with A fallback 1`] = `
<div>
  <span
    class="rt-reset rt-AvatarRoot rt-r-size-3 rt-variant-soft rounded-base w-[40px] h-[40px] "
    data-accent-color=""
  />
</div>
`;
